src/primitives/textInput/textInput.tsx

```diff
@@ -291,18 +297,20 @@ export const TextInput = forwardRef(function TextInput(
           style={CLEAR_BUTTON_BOX_STYLE}
           tone={customValidity ? 'critical' : 'inherit'}
         >
-          <Button
-            aria-label="Clear"
-            data-qa="clear-button"
-            fontSize={fontSize}
-            icon={CloseIcon}
-            mode="bleed"
-            padding={clearButtonPadding}
-            radius={radius}
-            {...clearButtonProps}
-            onClick={handleClearClick}
-            onMouseDown={handleClearMouseDown}
-          />
+          <RightFlex>
```

Comment:
```suggestion
          <Flex align="center" justify="center">
```

This should do the same as what `RightFlex` currently does, and saves us having to declare another styled component